chia/wallet/wallet_state_manager.py

```diff
@@ -1034,12 +1034,10 @@ async def new_coin_state(
         used_up_to = -1
         ph_to_index_cache: LRUCache = LRUCache(100)
 
-        local_records: List[Optional[WalletCoinRecord]] = await self.coin_store.get_coin_records(
-            [st.coin.name() for st in coin_states]
-        )
+        local_records = await self.coin_store.get_coin_records([st.coin.name() for st in coin_states])
 
-        assert len(local_records) == len(coin_states)
```

Comment:
this assert seems relevant still, no?

Reply:
No because we not longer include the `None` entries. If a local record isn't found the assertion isn't true anymore.